waffle-chai/src/matchers/ethCalled/calledOnContract.ts

```diff
@@ -0,0 +1,28 @@
+import {validateExpectArguments, validateFnName} from './calledOnContractValidators';
+import {RecordedCall} from '@ethereum-waffle/provider';
+
+export function supportCalledOnContract(Assertion: Chai.AssertionStatic) {
+  Assertion.addMethod('calledOnContract', function (this: any, contract: any) {
+    const fnName = this._obj;
+
+    validateExpectArguments(contract);
+    if (fnName !== undefined) {
+      validateFnName(fnName, contract);
+    }
+
+    const fnSighash = contract.interface.functions[fnName].sighash;
+    const callHistory = contract.provider.callHistory;
+
+    this.assert(
+      callHistory.some((call: RecordedCall) => {
```

Comment:
why no one line and without the explicit type and an additional variable
```ts 
contract.provider.callHistory.some(
  call => call.address === contract.address && call.data.startsWith(fnSighash)
)
```

Reply:
@sz-piotr, I added type because without it there is TS warning. Should I leave RecordedCall type?

Reply:
Oh it is actually a deeper problem. Typescript does not understand `validateExpectArguments` and thinks contract is `any`. You should validate the contract directly:
```
validateContract(contract);
validateMockProvider(contract.provider);
```

Reply:
Then you can remove the `RecordedCall` type from the callback